wallet/udb/txmined.go

```diff
@@ -402,19 +405,37 @@ func (s *Store) IsMissingMainChainCFilters(dbtx walletdb.ReadTx) bool {
 	return len(v) != 1 || v[0] == 0
 }
 
+// SetMissingMainChainCFilters sets whether we have all of the main chain
+// cfilters. Should be used to set missing to false if the wallet birthday is
+// moved back in time.
+func (s *Store) SetMissingMainChainCFilters(dbtx walletdb.ReadWriteTx, have bool) error {
```

Comment:
instead of a boolean value, we could just record the earliest block height that we have saved cfilters from.  If that's 0.

Reply:
Sorry, do you mean replace the boolean with a block number? I guess it would need an upgrade in the case of false.

Reply:
I looked into doing this but unsure if it simplifies the process. If we only save a number, and zero has the meaning that we have them all, then we loose the value of having the number except while downloading. I think the current code handles that well enough though, and it is only when changing the birthday. It may also require a new db version to replace the current boolean if I understand correctly.

Reply:
true, i think two fields should be used, the height/block of the first cfilter, and the last cfilter

Reply:
Am I understanding correctly in that we should replace the bool or add more. I'll tinker some more.

Reply:
The birthday would be where we want to save from, but it could be earlier. So another db value for where the actual first cfilter should be, and where the last one is. The last one will be the head block filter once we are synced, will be update that every block?